src/decoder/symphonia.rs

```diff
@@ -301,10 +301,13 @@ fn skip_back_a_tiny_bit(
     }: Time,
 ) -> Time {
     frac -= 0.0001;
```

Comment:
Maybe
```rust
if frac < 0.0001 {
} else {
}
```?

Reply:
its never called at the beginning of the source. This handles seeking beyond a source.

Reply:
so this is good to merge right?

Reply:
If it fixes the problem - yes, we can merge. Since it is already an improvement.
We can handle other cases later if necessary.
The source itself can be less than a second long. The check compares with < 1ms overlap, but tick-back is 0.1ms. what about overlaps in range 0.1...1.0 ms?  I wonder if there are any symphonia functions to work with their timestamps or ways to convert those from/to Duration.

Reply:
> The source itself can be less than a second long

ahh damn, I did not think of that. Thanks for clarifying. No I agree now, lets fix this properly only then merge. This will need a comment in the code explaining this case or a future refactor might throw the code out.

Reply:
For which codecs is it necessary to `skip_back_a_teeny_bit`? Code comment says that some codecs need it, but I can’t remember running into it before.

Reply:
I placed this comment in the code: 
```rust
        let time = if seek_beyond_end {
            let time = self.total_duration.expect("if guarantees this is Some");
            skip_back_a_tiny_bit(time) // some decoders can only seek to just before the end
        } else {
            pos.as_secs_f64().into()
        };
```

As I remember symphonia does not seek if the time is beyond the length of the track. In rodio we have defined seek to saturate at the track length. If symphonia now saturates we can remove this.

Reply:
When you seek to or beyond the end, you get an `UnexpectedEof` which is anyway what you get when the stream ends.

Reply:
What do you mean with saturating? I assume the behavior should be that the iterator returns `None`. Is that also what you mean?

I would make (and hav made) the Symphonia decoder such that it handles _this specific end-of-stream_ `UnexpectedEof` that way.

Reply:
> What do you mean with saturating?

see the docs: https://docs.rs/rodio/latest/rodio/source/trait.Source.html#method.try_seek

*As long as the duration of the source is known seek is guaranteed to saturate at the end of the source. For example given a source that reports a total duration of 42 seconds calling try_seek() with 60 seconds as argument will seek to 42 seconds.*

> I would make (and have made) the Symphonia decoder such that it handles this specific end-of-stream UnexpectedEof that way.

Yeah that's way simpler, go ahead and do that :+1: Maybe I was worried about skipping back? but that's not a feature nor can we make it one.